src/server/twitterSearch.js

```diff
@@ -0,0 +1,123 @@
+  return db.query(query, { 'params': flattenImmutableObject(objectToFlatten) });
+};
+
+const upsertTweeter = (db, tweeter) => {
+  return runQueryOnImmutableObject(
+    db,
+    'UPDATE tweeter SET id=:id, name=:name, handle=:handle UPSERT WHERE handle=:handle',
+    tweeter);
+};
+
+const upsertTweet = (db, tweet) => {
+  return runQueryOnImmutableObject(
+    db,
+    'UPDATE tweet SET id=:id, content=:content, date=:date, likes=:likes, retweets=:retweets UPSERT WHERE id=:id',
+    tweet);
+};
+
+const upsertHashtag = (db, hashtag) => {
+  return runQueryOnImmutableObject(
+    db,
+    'UPDATE hashtag SET content=:content UPSERT WHERE content=:content',
+    hashtag);
+};
+
+const linkTweeterToTweet = (db, tweeter, tweet) => {
+  return db.query(
+    'CREATE EDGE TWEETED FROM (SELECT FROM tweeter WHERE id = :tweeterId) TO (SELECT FROM tweet WHERE id = :tweetId)',
+    {
+      'params': {
+        'tweetId': tweet.id(),
+        'tweeterId': tweeter.id(),
+      },
+    });
+};
+
+const linkTweetToHashtag = (db, tweet, hashtag) => {
+  return db.query(
+    'CREATE EDGE HAS_HASHTAG FROM (SELECT FROM tweet WHERE id = :tweetId) TO (SELECT FROM hashtag WHERE content = :hashtagContent)',
+    {
+      'params': {
+        'tweetId': tweet.id(),
+        'hashtagContent': hashtag.content(),
+      },
+    });
+};
+
+const linkTweetToTweeterViaMention = (db, tweet, mentionedTweeter) => {
+  return db.query(
+    'CREATE EDGE MENTIONS FROM (SELECT FROM tweet WHERE id = :tweetId) TO (SELECT FROM tweeter WHERE id = :mentionedTweeterId)',
+    {
+      'params': {
+        'tweetId': tweet.id(),
+        'mentionedTweeterId': mentionedTweeter.id(),
+      },
+    });
+};
+
+const makeTweeterFromRaw = (raw) => {
+  return Builders.TweeterBuilder()
+    .id(raw.id)
+    .name(raw.name)
+    .handle(raw.screen_name)
+    .build();
+};
+
+export const searchAndSave = (res, query) => {
+  T.get('search/tweets', { 'q': query, count: 300 }, function (err, result, response) {
+    let count = 0;
+    result.statuses.forEach((tweetRaw) => {
+      const userRaw = tweetRaw.user;
+
+      const tweet = Builders.TweetBuilder()
+        .id(tweetRaw.id)
+        .content(tweetRaw.text)
+        .date(moment(new Date(tweetRaw.created_at)).format('YYYY-MM-DD HH:mm:ss'))
+        .likes(tweetRaw.favourite_count || 0)
+        .retweets(tweetRaw.retweet_count || 0)
+        .build();
+
+      const tweeter = makeTweeterFromRaw(userRaw);
+
+      upsertTweeter(db, tweeter).then((result) => {
+        upsertTweet(db, tweet).then((result) => {
+          linkTweeterToTweet(db, tweeter, tweet);
+        });
+      });
+
+      tweetRaw.entities.hashtags.forEach((hashtagRaw) => {
+        const hashtag = Builders.HashtagBuilder().content(hashtagRaw.text.toLowerCase()).build();
+        upsertHashtag(db, hashtag).then((result) => {
+          linkTweetToHashtag(db, tweet, hashtag);
+        });
+      });
+
+      tweetRaw.entities.user_mentions.forEach((mentionRaw) => {

```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Identifier 'user_mentions' is not in camel case.](https://www.codacy.com/app/winwardo/solid-octo-disco/file/2174002023/issues/source?bid=3148782&fileBranchId=3159920#l113)